### PR TITLE
GEODE-8885: move hincrby command to supported

### PR DESCRIPTION
**Pull request by nonbinaryprogrammer**
- add unit/integration tests
- add dunit concurrency test
- update docs
- update SupportedCommandsJUnitTest

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [x] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [x] Is your initial contribution a single, squashed commit?

- [x] Does `gradlew build` run cleanly?

- [x] Have you written or updated unit tests to verify your changes?

- [x] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, check Concourse for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: geode-redis/src/distributedTest/java/org/apache/geode/redis/internal/executor/hash/HIncrByDUnitTest.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+  @Test
+  public void testConcurrentHIncrBy_performsAllIncrBys() {
+    String key = "key";
+    String field = "field";
+    AtomicInteger expectedValue = new AtomicInteger(0);
+
+    jedis1.hset(key, field, "0");
+
+    new ConcurrentLoopingThreads(NUM_ITERATIONS,
```
Comment (jdeppe-pivotal) [inline geode-redis/src/distributedTest/java/org/apache/geode/redis/internal/executor/hash/HIncrByDUnitTest.java:95]: This is missing `.run()` to execute it. Consider moving this to integration tests and that way native redis will also run it.

File: geode-redis/src/integrationTest/java/org/apache/geode/redis/internal/executor/hash/AbstractHashesIntegrationTest.java [hunk 2]
```diff
@@ -275,32 +275,49 @@ public void testHkeys() {
   }
 
   @Test
-  public void testHIncrBy() {
-    String key = "key";
-    String field = "field";
+  public void testHIncrBy_returnsErrorMessageForWrongNumberOfParameters() {
```
Comment (jdeppe-pivotal) [inline geode-redis/src/integrationTest/java/org/apache/geode/redis/internal/executor/hash/AbstractHashesIntegrationTest.java:278]: Thanks for making these tests method names more explicit!

**Review by jdeppe-pivotal — CHANGES_REQUESTED** [review]

**Review by jdeppe-pivotal — APPROVED** [review]

File: geode-redis/src/integrationTest/java/org/apache/geode/redis/internal/executor/hash/AbstractHashesIntegrationTest.java [hunk 3]
```diff
@@ -48,6 +50,7 @@
   private Jedis jedis;
   private Jedis jedis2;
   private static int ITERATION_COUNT = 4000;
+  private static int NUM_ITERATIONS = 1000;
```
Comment (ringles) [inline geode-redis/src/integrationTest/java/org/apache/geode/redis/internal/executor/hash/AbstractHashesIntegrationTest.java:53]: Having a constant named "ITERATION_COUNT" right next to a constant named "NUM_ITERATIONS" feels weird. At least one of them should be renamed to be a bit more specific.

Comment (nonbinaryprogrammer) [reply geode-redis/src/integrationTest/java/org/apache/geode/redis/internal/executor/hash/AbstractHashesIntegrationTest.java]: 👍 good call. NUM_ITERATIONS was only used once so I just got rid of it

**Pull request merged**